### DISCUSSION TITLE
Question about v1.29's no CLBLAST for K-quants

### DISCUSSION THREAD

**Discussion by ghost** [LostRuins/koboldcpp#219] [Q&A]
So I understand that the new K-quantization is only supported by openblas for now. Without CLblast does that also mean no GPU layers offloading?

**Comment by LostRuins** [comment]
That is correct. Until the dequantize and mat_mul kernels are implemented, K-quants will not work with CLBlast.

You can also follow this issue https://github.com/ggerganov/llama.cpp/issues/1725 

Someone will have to come up with the implementation as I am not familiar enough with CL kernels to do them correctly. I'm not sure why the person who added k-quants didn't add a CL implementation as well.

**Reply by ghost** [reply]
I think I could have phrased my question better. I was wondering if we can (for now) still keep GPU layer offloading without CLBlast

**Reply by LostRuins** [reply]
Unfortunately no, CLBlast is required for GPU offloading.

**Comment by LostRuins** [comment]
If anyone is able to write CL kernels, contributions (either here or preferably upstream) are greatly welcome

**Reply by cgessai** [reply]
I can't tell, is [this announcement](https://www.reddit.com/r/LocalLLaMA/comments/147z6as/llamacpp_just_got_full_cuda_acceleration_and_now/) from today a solution to the issue we're talking about?

**Reply by Thellton** [reply]
> I can't tell, is [this announcement](https://www.reddit.com/r/LocalLLaMA/comments/147z6as/llamacpp_just_got_full_cuda_acceleration_and_now/) from today a solution to the issue we're talking about?

Unfortunately not. CUDA is a proprietary API that is only available on Nvidia GPUs. this will likely end up coming down stream in time to Koboldcpp due to it being a fork of LLaMAcpp so for anybody with a Nvidia GPU, they've got that to look forward to. for those of us with AMD and Intel GPUs... we're just going to have to watch the [q2_K](https://github.com/ggerganov/llama.cpp/issues/1804) and [q3_K](https://github.com/ggerganov/llama.cpp/issues/1725) issues over at LLaMAcpp for movements :(

**Reply by cgessai** [reply]
Ah, thank you for the additional clarification!

**Comment by LostRuins** [comment]
Implemented in newest version.
